sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -62,13 +65,29 @@ public TextAnalyticsClient(Uri endpoint, TokenCredential credential, TextAnalyti
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(options, nameof(options));
 
-            string defaultScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? TextAnalyticsAudience.AzurePublicCloud : options.Audience)}/.default";
-            _baseUri = endpoint;
-            _clientDiagnostics = new TextAnalyticsClientDiagnostics(options);
-            _options = options;
+            var authorizationScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? TextAnalyticsAudience.AzurePublicCloud : options.Audience)}/.default";
 
-            var pipeline = HttpPipelineBuilder.Build(options, new BearerTokenAuthenticationPolicy(credential, defaultScope));
-            _languageRestClient = new MicrosoftCognitiveLanguageServiceRestClient(_clientDiagnostics, pipeline, endpoint.AbsoluteUri, TextAnalyticsClientOptions.GetVersionString(options.Version));
+            _serviceClient = options.Version switch
+            {
+                var version when s_legacyServiceVersions.Contains(version) =>
+                    new LegacyServiceClient(
+                        endpoint,
+                        credential,
+                        options.DefaultLanguage,
+                        options.DefaultCountryHint,
```

Comment:
curious on why the need to pass these 2 properties directly, instead of getting their values from `options` in the respective clients.
My concern is if in the future we keep adding defaults to the library, we are going to have to add more parameters to the method

Reply:
There's context in the previous PR, but the short version is that I was on the fence about how proper it was to expose the `TextAnalyticsClientOptions` to a class that wasn't `TextAnalyticsClient` - it just feels like a leaky abstraction.    

I'm open to changing it, it's not something that I felt strongly about.

Reply:
Apologies I missed that in the last review.
I don't feel strongly about it either. We can leave it as is and if it ever is a problem, then change it

Reply:
It was a good call-out; I ended up needing `ClientOptions` anyway, which I wasn't accounting for initially.   I updated to just pass the options.